media_remover/src/test/scala/MediaNotRequiredMessageProcessorSpec.scala

```diff
@@ -258,124 +498,1657 @@ class MediaNotRequiredMessageProcessorSpec extends Specification with Mockito {
+      pendingDeletionRecordDAO.writeRecord(any) returns Future(234)
+
+      implicit val vidispineCommunicator = mock[VidispineCommunicator]
+
+      implicit val mat:Materializer = mock[Materializer]
+      implicit val sys:ActorSystem = mock[ActorSystem]
+      implicit val mockBuilder = mock[MXSConnectionBuilderImpl]
+      implicit val mockS3ObjectChecker = mock[S3ObjectChecker]
+      implicit val mockChecksumChecker = mock[ChecksumChecker]
+      val mockAssetFolderLookup = mock[AssetFolderLookup]
+
+      val toTest = new MediaNotRequiredMessageProcessor(mockAssetFolderLookup)
+
+      val msgContent =
+        """{
+          |"mediaTier": "NEARLINE",
+          |"projectIds": ["22"],
+          |"nearlineId": "1",
+          |"vidispineItemId": "VX-151922",
+          |"mediaCategory": "Deliverables"
+          |}""".stripMargin
+
+      val msgObj = io.circe.parser.parse(msgContent).flatMap(_.as[OnlineOutputMessage]).right.get
+
+      val result = Try {
+        Await.result(toTest.storeDeletionPending(msgObj), 2.seconds)
+      }
+
+      result must beAFailedTry
+      result.failed.get must beAnInstanceOf[RuntimeException]
+      println(s"sdp-a-result: $result")
+      result.failed.get.getMessage mustEqual "Cannot store PendingDeletion record for item without filepath"
+    }
+
+
+    "store new record for NEARLINE if no record found" in {
+      val mockMsgFramework = mock[MessageProcessingFramework]
+      implicit val pendingDeletionRecordDAO :PendingDeletionRecordDAO = mock[PendingDeletionRecordDAO]
+      pendingDeletionRecordDAO.findByNearlineIdForNEARLINE(any) returns Future(None)
+      pendingDeletionRecordDAO.writeRecord(any) returns Future(234)
+
+      implicit val vidispineCommunicator = mock[VidispineCommunicator]
+
+      implicit val mat:Materializer = mock[Materializer]
+      implicit val sys:ActorSystem = mock[ActorSystem]
+      implicit val mockBuilder = mock[MXSConnectionBuilderImpl]
+      implicit val mockS3ObjectChecker = mock[S3ObjectChecker]
+      implicit val mockChecksumChecker = mock[ChecksumChecker]
+      val mockAssetFolderLookup = mock[AssetFolderLookup]
+
+      val toTest = new MediaNotRequiredMessageProcessor(mockAssetFolderLookup)
+
+      val msgContent =
+        """{
+          |"mediaTier": "NEARLINE",
+          |"projectIds": ["22"],
+          |"originalFilePath": "/srv/Multimedia2/Media Production/Assets/Multimedia_Reactive_News_and_Sport/Reactive_News_Explainers_2022/monika_cvorak_MH_Investigation/Footage Vera Productions/2022-03-18_MH.mp4",
+          |"nearlineId": "1",
+          |"vidispineItemId": "VX-151922",
+          |"mediaCategory": "Deliverables"
+          |}""".stripMargin
+
+      val msgObj = io.circe.parser.parse(msgContent).flatMap(_.as[OnlineOutputMessage]).right.get
+
+      val result = Try {
+        Await.result(toTest.storeDeletionPending(msgObj), 2.seconds)
+      }
+
```

Comment:
you should probably put an assertion of the mock call in here, something like
```scala
there was one(pendingDeletionRecordDAO).writeRecord(msgObj)
```

(assuming that msgObj is what should actually get written to the db)

Reply:
Sure thing! (it's not, but I get the drift)